client/src/main/java/io/pravega/client/stream/impl/TransactionalEventStreamWriterImpl.java

```diff
@@ -177,8 +179,10 @@ private void throwIfClosed() throws TxnFailedException {
         UUID txnId = txnSegments.getTxnId();
         Map<Segment, SegmentTransaction<Type>> transactions = new HashMap<>();
         for (Segment s : txnSegments.getSteamSegments().getSegments()) {
+            DelegationTokenProvider tokenProvider = DelegationTokenProviderFactory.create(
```

Comment:
Again this is one per-segment and not reused between calls.

Reply:
I've made this change in the latest commit.

Reply:
It's there in the latest code. 

https://github.com/pravega/pravega/blob/5d21e6a484657f8c9dc0a6c74d7337e8560bb19b/client/src/main/java/io/pravega/client/stream/impl/TransactionalEventStreamWriterImpl.java#L181-L186